client/apps/settings/client/curations/display/components/canvas_images.jsx

```diff
@@ -19,13 +19,8 @@ export class CanvasImages extends React.Component {
   }
 
   onImageInputChange = (key, value, imgIndex) => {
-    let newValue
-    if (this.props.campaign.canvas.layout === 'slideshow') {
-      newValue = this.onSlideshowImageChange(imgIndex, value)
-    } else {
-      newValue = value.length ? [{url: value}] : []
-    }
-    this.props.onChange(key, newValue, this.props.index)
+    const assets = this.updateImageUrls(imgIndex, value)
+    this.props.onChange(key, assets, this.props.index)
```

Comment:
Minor, but its a good habit to get into is if you use a prop or state value more than once to pluck it out to keep things DRY:

```javascript 
const { 
  onChange, 
  index 
} = this.props 

const assets = this.updateImageUrls(imgIndex, value)
onChange(key, assets, index)
```